File: easy/125.java
class Solution {
    public boolean isPalindrome(String s) {
        String ans = s.toLowerCase().trim();
        int i = 0;
        int j = ans.length() - 1;
        while (i < j){
            if (!((ans.charAt(i) >= 48 && ans.charAt(i) <= 57) || (ans.charAt(i) >= 97 && ans.charAt(i) <= 122))){
                ++i;
            } else if (!((ans.charAt(j) >= 48 && ans.charAt(j) <= 57) || (ans.charAt(j) >= 97 && ans.charAt(j) <= 122))){
                --j;
            } else {
                if (ans.charAt(i) != ans.charAt(j)){
                    return false;
                }
                ++i;
                --j;
            }
        }
        return true;
    }
}

File: easy/125.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        t = s.lower().strip()
        i = 0
        j = len(t) - 1
        while i < j:
            if not t[i].isalnum():
                i += 1
            elif not t[j].isalnum():
                j -= 1
            else:
                if t[i] != t[j]:
                    return False
                i += 1
                j -= 1
        return True
